uaclient/util.py

```diff
@@ -100,7 +102,43 @@ def object_hook(o):
         return o
 
 
-def apply_series_overrides(
+def _get_override_weight(
+    override_selector: Dict[str, str], selector_values: Dict[str, str]
+) -> int:
+    override_weight = 0
+    for selector, value in override_selector.items():
+        if (selector, value) not in selector_values.items():
+            return 0
+        override_weight += OVERRIDE_SELECTOR_WEIGHTS[selector]
+
+    return override_weight
+
+
+def _select_overrides(
+    entitlement: Dict[str, Any], series_name: str, cloud_type: str
+) -> Dict[int, Dict[str, Any]]:
+    overrides = {}
+
+    selector_values = {"series": series_name, "cloud": cloud_type}
+
+    series_overrides = entitlement.pop("series", {}).pop(series_name, {})
```

Comment:
Aren't you modifying the global `entitlement` dict here by performing `pop`. I am just worried if might need that information later in the process

Reply:
Right this aligns with the old implementation.

Not for this PR, but I think a worthwhile refactor would be to make this more "functional". So it wouldn't modify the argument and would return a new object with overrides applied